app/Main.hs

```diff
@@ -58,5 +58,33 @@ eval env (App t u) =
             Just value -> eval ((arg, value) : env') expr
         _                            -> Nothing
 
+checkShadowing :: [Identifier] -> Expr -> [Identifier]
+checkShadowing args (Abs arg expr) 
+  | arg !! 0 == '_' = nested
```

Comment:
Why not just `arg == '_'`?
Otherwise we allow all identifiers beginning with `_`.
NB: `!!` is unsafe 😕

Reply:
this is not a bug but a feature 😄:

Reply:
okay, so be it. we'll remove it if it gets too confusing 👍